diff --git a/bfd/acinclude.m4 b/bfd/acinclude.m4
index d9813bd..5ecaebf 100644
--- a/bfd/acinclude.m4
+++ b/bfd/acinclude.m4
@@ -61,7 +61,8 @@ AC_DEFUN([AM_INSTALL_LIBBFD],
   # libbfd.a is a host library containing target dependent code
   bfdlibdir='$(libdir)'
   bfdincludedir='$(includedir)'
-  if test "${host}" != "${target}"; then
+### Force non-native build
+  if test "${host}" != "${target}" || true; then
     bfdlibdir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/lib'
     bfdincludedir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/include'
   fi
diff --git a/bfd/configure b/bfd/configure
index d587593..d95999c 100755
--- a/bfd/configure
+++ b/bfd/configure
@@ -12298,7 +12298,8 @@ fi
   # libbfd.a is a host library containing target dependent code
   bfdlibdir='$(libdir)'
   bfdincludedir='$(includedir)'
-  if test "${host}" != "${target}"; then
+### Force non-native build
+  if test "${host}" != "${target}" || true; then
     bfdlibdir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/lib'
     bfdincludedir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/include'
   fi
@@ -13821,7 +13822,8 @@ COREFILE=
 COREFLAG=
 CORE_HEADER=
 TRAD_HEADER=
-if test "${target}" = "${host}"; then
+### Force non-native build
+if test "${target}" = "${host}" && false; then
   case "${host}" in
   alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu | alpha*-*-*vms*)
 	COREFILE=''
diff --git a/bfd/configure.in b/bfd/configure.in
index eaba44c..3b468dc 100644
--- a/bfd/configure.in
+++ b/bfd/configure.in
@@ -221,7 +221,8 @@ COREFILE=
 COREFLAG=
 CORE_HEADER=
 TRAD_HEADER=
-if test "${target}" = "${host}"; then
+### Force non-native build
+if test "${target}" = "${host}" && false; then
   case "${host}" in
   alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu | alpha*-*-*vms*)
 	COREFILE=''
diff --git a/configure b/configure
index 8a146e2..a0c12a1 100755
--- a/configure
+++ b/configure
@@ -2274,7 +2274,8 @@ esac
 
 
 
-test "$host_noncanonical" = "$target_noncanonical" &&
+### Force non-native build
+test "$host_noncanonical" = "$target_noncanonical" && false &&
   test "$program_prefix$program_suffix$program_transform_name" = \
     NONENONEs,x,x, &&
   program_transform_name=s,y,y,
@@ -2714,7 +2715,8 @@ appdirs=""
 
 # Define is_cross_compiler to save on calls to 'test'.
 is_cross_compiler=
-if test x"${host}" = x"${target}" ; then
+### Force non-native build
+if test x"${host}" = x"${target}" && false; then
   is_cross_compiler=no
 else
   is_cross_compiler=yes
diff --git a/configure.ac b/configure.ac
index 68208dc..72ac0e7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -71,7 +71,8 @@ dnl Autoconf 2.5x and later will set a default program prefix if
 dnl --target was used, even if it was the same as --host.  Disable
 dnl that behavior.  This must be done before AC_CANONICAL_SYSTEM
 dnl to take effect.
-test "$host_noncanonical" = "$target_noncanonical" &&
+### Force non-native build
+test "$host_noncanonical" = "$target_noncanonical" && false &&
   test "$program_prefix$program_suffix$program_transform_name" = \
     NONENONEs,x,x, &&
   program_transform_name=s,y,y,
@@ -205,7 +206,8 @@ appdirs=""
 
 # Define is_cross_compiler to save on calls to 'test'.
 is_cross_compiler=
-if test x"${host}" = x"${target}" ; then
+### Force non-native build
+if test x"${host}" = x"${target}" && false; then
   is_cross_compiler=no
 else
   is_cross_compiler=yes
diff --git a/gas/configure b/gas/configure
index 52bc0cd..b4d185a 100755
--- a/gas/configure
+++ b/gas/configure
@@ -13176,9 +13176,10 @@ done
 
 # Put this here so that autoconf's "cross-compiling" message doesn't confuse
 # people who are not cross-compiling but are compiling cross-assemblers.
+### Force non-native build
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether compiling a cross-assembler" >&5
 $as_echo_n "checking whether compiling a cross-assembler... " >&6; }
-if test "${host}" = "${target}"; then
+if test "${host}" = "${target}" && false; then
   cross_gas=no
 else
   cross_gas=yes
diff --git a/gas/configure.in b/gas/configure.in
index 69f5fcb..0cbe58f 100644
--- a/gas/configure.in
+++ b/gas/configure.in
@@ -630,8 +630,9 @@ AC_CHECK_HEADERS(string.h stdlib.h memory.h strings.h unistd.h errno.h sys/types
 
 # Put this here so that autoconf's "cross-compiling" message doesn't confuse
 # people who are not cross-compiling but are compiling cross-assemblers.
+### Force non-native build
 AC_MSG_CHECKING(whether compiling a cross-assembler)
-if test "${host}" = "${target}"; then
+if test "${host}" = "${target}" && false; then
   cross_gas=no
 else
   cross_gas=yes
diff --git a/gold/configure b/gold/configure
index 5a7268d..8697d5e 100755
--- a/gold/configure
+++ b/gold/configure
@@ -6080,7 +6080,8 @@ $as_echo "#define AC_APPLE_UNIVERSAL_BUILD 1" >>confdefs.h
 
 
 
- if test "x$target_alias" = "x" -o "x$host_alias" = "x$target_alias"; then
+### Force non-native build
+ if test "x$target_alias" = "x" -o "x$host_alias" = "x$target_alias" -a false; then
   NATIVE_LINKER_TRUE=
   NATIVE_LINKER_FALSE='#'
 else
diff --git a/gold/configure.ac b/gold/configure.ac
index 7757d8c..16a2536 100644
--- a/gold/configure.ac
+++ b/gold/configure.ac
@@ -254,8 +254,9 @@ AC_C_BIGENDIAN
 
 AC_EXEEXT
 
+### Force non-native build
 AM_CONDITIONAL(NATIVE_LINKER,
-  test "x$target_alias" = "x" -o "x$host_alias" = "x$target_alias")
+  test "x$target_alias" = "x" -o "x$host_alias" = "x$target_alias" -a false)
 AM_CONDITIONAL(GCC, test "$GCC" = yes)
 
 AM_CONDITIONAL(NATIVE_OR_CROSS_LINKER,
diff --git a/ld/emulparams/elf32bmipn32-defs.sh b/ld/emulparams/elf32bmipn32-defs.sh
index 45bfd5d..b8fc9e4 100644
--- a/ld/emulparams/elf32bmipn32-defs.sh
+++ b/ld/emulparams/elf32bmipn32-defs.sh
@@ -19,7 +19,8 @@ elf64*) ELFSIZE=64 ;;
 *) echo $0: unhandled emulation $EMULATION_NAME >&2; exit 1 ;;
 esac
 
-if test `echo "$host" | sed -e s/64//` = `echo "$target" | sed -e s/64//`; then
+### Force non-native build
+if test `echo "$host" | sed -e s/64//` = `echo "$target" | sed -e s/64//` && false; then
   case " $EMULATION_LIBPATH " in
     *" ${EMULATION_NAME} "*)
       NATIVE=yes
diff --git a/ld/emulparams/elf32ppccommon.sh b/ld/emulparams/elf32ppccommon.sh
index 15cb3b1..9eef176 100644
--- a/ld/emulparams/elf32ppccommon.sh
+++ b/ld/emulparams/elf32ppccommon.sh
@@ -36,7 +36,8 @@ OTHER_GOT_RELOC_SECTIONS="
 
 # Treat a host that matches the target with the possible exception of "64"
 # in the name as if it were native.
-if test `echo "$host" | sed -e s/64//` = `echo "$target" | sed -e s/64//`; then
+### Force non-native build
+if test `echo "$host" | sed -e s/64//` = `echo "$target" | sed -e s/64//` && false; then
   case " $EMULATION_LIBPATH " in
     *" ${EMULATION_NAME} "*)
       NATIVE=yes
diff --git a/ld/emulparams/elf64ppc.sh b/ld/emulparams/elf64ppc.sh
index c9337ea..4b7b54f 100644
--- a/ld/emulparams/elf64ppc.sh
+++ b/ld/emulparams/elf64ppc.sh
@@ -41,7 +41,8 @@ OTHER_READWRITE_SECTIONS="
 
 # Treat a host that matches the target with the possible exception of "64"
 # in the name as if it were native.
-if test `echo "$host" | sed -e s/64//` = `echo "$target" | sed -e s/64//`; then
+### Force non-native build
+if test `echo "$host" | sed -e s/64//` = `echo "$target" | sed -e s/64//` && false; then
   case " $EMULATION_LIBPATH " in
     *" ${EMULATION_NAME} "*)
       NATIVE=yes
diff --git a/ld/emulparams/elf_l1om.sh b/ld/emulparams/elf_l1om.sh
index 1af2360..6f0c04c 100644
--- a/ld/emulparams/elf_l1om.sh
+++ b/ld/emulparams/elf_l1om.sh
@@ -16,7 +16,8 @@ NO_SMALL_DATA=yes
 LARGE_SECTIONS=yes
 SEPARATE_GOTPLT=24
 
-if [ "x${host}" = "x${target}" ]; then
+### Force non-native build
+if [ "x${host}" = "x${target}" ] && false; then
   case " $EMULATION_LIBPATH " in
     *" ${EMULATION_NAME} "*)
       NATIVE=yes
diff --git a/ld/emulparams/elf_x86_64.sh b/ld/emulparams/elf_x86_64.sh
index 0e93ad8..e114f2f 100644
--- a/ld/emulparams/elf_x86_64.sh
+++ b/ld/emulparams/elf_x86_64.sh
@@ -16,7 +16,8 @@ LARGE_SECTIONS=yes
 SEPARATE_GOTPLT=24
 IREL_IN_PLT=
 
-if [ "x${host}" = "x${target}" ]; then
+### Force non-native build
+if [ "x${host}" = "x${target}" ] && false; then
   case " $EMULATION_LIBPATH " in
     *" ${EMULATION_NAME} "*)
       NATIVE=yes
diff --git a/ld/emultempl/sunos.em b/ld/emultempl/sunos.em
index 567b8e2..4e3a46f 100644
--- a/ld/emultempl/sunos.em
+++ b/ld/emultempl/sunos.em
@@ -98,7 +98,8 @@ static void
 gld${EMULATION_NAME}_set_symbols (void)
 {
 EOF
-if [ "x${host}" = "x${target}" ] ; then
+### Force non-native build
+if [ "x${host}" = "x${target}" ] && false; then
   case " ${EMULATION_LIBPATH} " in
   *" ${EMULATION_NAME} "*)
 fragment <<EOF
@@ -463,7 +464,8 @@ gld${EMULATION_NAME}_after_open (void)
 	    continue;
 	}
 EOF
-if [ "x${host}" = "x${target}" ] ; then
+### Force non-native build
+if [ "x${host}" = "x${target}" ] && false ; then
   case " ${EMULATION_LIBPATH} " in
   *" ${EMULATION_NAME} "*)
 fragment <<EOF
diff --git a/ld/genscripts.sh b/ld/genscripts.sh
index 61981bd..4f49168 100755
--- a/ld/genscripts.sh
+++ b/ld/genscripts.sh
@@ -122,9 +122,10 @@ fi
 # may already have been set by the emulparams file, but that's OK
 # (it'll just get set to "yes" twice).
 
+### Force non-native build
 case " $EMULATION_LIBPATH " in
   *" ${EMULATION_NAME} "*)
-    if [ "x${host}" = "x${target}" ] ; then
+    if [ "x${host}" = "x${target}" ] && false; then
       NATIVE=yes
       USE_LIBPATH=yes
     elif [ "x${use_sysroot}" = "xyes" ] ; then
diff --git a/opcodes/configure b/opcodes/configure
index 83ac9ae..e353992 100755
--- a/opcodes/configure
+++ b/opcodes/configure
@@ -11638,7 +11638,8 @@ fi
   # libbfd.a is a host library containing target dependent code
   bfdlibdir='$(libdir)'
   bfdincludedir='$(includedir)'
-  if test "${host}" != "${target}"; then
+### Force non-native build
+  if test "${host}" != "${target}" || true; then
     bfdlibdir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/lib'
     bfdincludedir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/include'
   fi
