diff -ruNa binutils-2.21.51.20110421.orig//bfd/Makefile.in binutils-2.21.51.20110421/bfd/Makefile.in
--- binutils-2.21.51.20110421.orig//bfd/Makefile.in	2011-08-10 10:52:06.737372000 +0900
+++ binutils-2.21.51.20110421/bfd/Makefile.in	2011-08-10 11:16:21.663372000 +0900
@@ -323,7 +323,7 @@
 # RELEASE=y
 INCDIR = $(srcdir)/../include
 CSEARCH = -I. -I$(srcdir) -I$(INCDIR)
-SUBDIRS = doc po
+SUBDIRS = po
 bfddocdir = doc
 libbfd_la_LDFLAGS = $(am__append_1) -release `cat libtool-soversion` \
 	@SHARED_LDFLAGS@ $(am__empty)
@@ -1738,7 +1738,7 @@
 
 html-am:
 
-info: info-recursive
+#info: info-recursive
 
 info-am:
 
diff -ruNa binutils-2.21.51.20110421.orig//binutils/Makefile.in binutils-2.21.51.20110421/binutils/Makefile.in
--- binutils-2.21.51.20110421.orig//binutils/Makefile.in	2011-08-10 10:52:10.436372000 +0900
+++ binutils-2.21.51.20110421/binutils/Makefile.in	2011-08-10 11:22:16.121372001 +0900
@@ -378,7 +378,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = dejagnu no-dist foreign
 ACLOCAL_AMFLAGS = -I .. -I ../config -I ../bfd
-SUBDIRS = doc po
+SUBDIRS = po
 tooldir = $(exec_prefix)/$(target_alias)
 
 # Automake 1.10+ disables lex and yacc output file regeneration if
@@ -1086,7 +1086,7 @@
 
 html-am:
 
-info: info-recursive
+#info: info-recursive
 
 info-am:
 
diff -ruNa binutils-2.21.51.20110421.orig//etc/Makefile.in binutils-2.21.51.20110421/etc/Makefile.in
--- binutils-2.21.51.20110421.orig//etc/Makefile.in	2011-08-10 10:52:15.528372002 +0900
+++ binutils-2.21.51.20110421/etc/Makefile.in	2011-08-10 11:26:52.247372002 +0900
@@ -63,8 +63,8 @@
 PDFFILES = standards.pdf configure.pdf
 HTMLFILES = standards.html configure.html
 
-all: info
-install install-strip: install-info
+all:
+install install-strip: 
 
 uninstall:
 
diff -ruNa binutils-2.21.51.20110421.orig//gas/Makefile.in binutils-2.21.51.20110421/gas/Makefile.in
--- binutils-2.21.51.20110421.orig//gas/Makefile.in	2011-08-10 10:51:59.234372002 +0900
+++ binutils-2.21.51.20110421/gas/Makefile.in	2011-08-10 11:22:52.492372001 +0900
@@ -281,7 +281,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.11 dejagnu foreign no-dist
 ACLOCAL_AMFLAGS = -I .. -I ../config -I ../bfd
-SUBDIRS = doc po
+SUBDIRS = po
 tooldir = $(exec_prefix)/$(target_alias)
 
 # Automake 1.10+ disables lex and yacc output file regeneration if
diff -ruNa binutils-2.21.51.20110421.orig//gprof/Makefile.in binutils-2.21.51.20110421/gprof/Makefile.in
--- binutils-2.21.51.20110421.orig//gprof/Makefile.in	2011-08-10 10:52:09.935372001 +0900
+++ binutils-2.21.51.20110421/gprof/Makefile.in	2011-08-10 11:44:28.280371999 +0900
@@ -93,7 +93,7 @@
 	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
 SOURCES = $(gprof_SOURCES)
-INFO_DEPS = gprof.info
+INFO_DEPS = 
 am__TEXINFO_TEX_DIR = $(srcdir)/$(top_srcdir)/../texinfo
 DVIS = gprof.dvi
 PDFS = gprof.pdf
@@ -802,7 +802,7 @@
 check-am: all-am
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-recursive
-all-am: Makefile $(INFO_DEPS) $(PROGRAMS) $(MANS) $(HEADERS) gconfig.h
+all-am: Makefile $(PROGRAMS) $(MANS) $(HEADERS) gconfig.h
 installdirs: installdirs-recursive
 installdirs-am:
 	for dir in "$(DESTDIR)$(bindir)" "$(DESTDIR)$(infodir)" "$(DESTDIR)$(man1dir)"; do \
diff -ruNa binutils-2.21.51.20110421.orig//ld/Makefile.in binutils-2.21.51.20110421/ld/Makefile.in
--- binutils-2.21.51.20110421.orig//ld/Makefile.in	2011-08-10 10:52:11.370372002 +0900
+++ binutils-2.21.51.20110421/ld/Makefile.in	2011-08-10 11:44:58.931372000 +0900
@@ -117,7 +117,7 @@
 	--mode=compile $(YACC) $(YFLAGS) $(AM_YFLAGS)
 SOURCES = $(libldtestplug_la_SOURCES) $(ld_new_SOURCES) \
 	$(EXTRA_ld_new_SOURCES)
-INFO_DEPS = ld.info
+INFO_DEPS = 
 am__TEXINFO_TEX_DIR = $(srcdir)/$(top_srcdir)/../texinfo
 DVIS = ld.dvi
 PDFS = ld.pdf
@@ -1775,7 +1775,7 @@
 	$(MAKE) $(AM_MAKEFLAGS) check-DEJAGNU
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-recursive
-all-am: Makefile $(INFO_DEPS) $(LTLIBRARIES) $(PROGRAMS) $(MANS) \
+all-am: Makefile $(LTLIBRARIES) $(PROGRAMS) $(MANS) \
 		config.h
 installdirs: installdirs-recursive
 installdirs-am:
