diff --git a/bfd/elflink.c b/bfd/elflink.c
index 1f6c1a0..8daebe5 100644
--- a/bfd/elflink.c
+++ b/bfd/elflink.c
@@ -11721,10 +11721,15 @@
 static bfd_boolean
 elf_gc_sweep_symbol (struct elf_link_hash_entry *h, void *data)
 {
-  if ((h->root.type == bfd_link_hash_defined
-       || h->root.type == bfd_link_hash_defweak)
-      && !h->root.u.def.section->gc_mark
-      && !(h->root.u.def.section->owner->flags & DYNAMIC))
+  if (((h->root.type == bfd_link_hash_defined 
+	|| h->root.type == bfd_link_hash_defweak) 
+       && !h->root.u.def.section->gc_mark 
+       && (!(h->root.u.def.section->owner->flags & DYNAMIC) 
+	   || (h->plt.refcount <= 0 
+	       && h->got.refcount <= 0))) 
+      || (h->root.type == bfd_link_hash_undefined 
+	  && h->plt.refcount <= 0 
+	  && h->got.refcount <= 0)) 
     {
       struct elf_gc_sweep_symbol_info *inf =
           (struct elf_gc_sweep_symbol_info *) data;

diff --git a/ld/testsuite/ld-elf/pr13177.d b/ld/testsuite/ld-elf/pr13177.d
new file mode 100644
index 0000000..b58623a
--- /dev/null
+++ b/ld/testsuite/ld-elf/pr13177.d
@@ -0,0 +1,11 @@
+#source: pr13177.s 
+#ld: --gc-sections -shared 
+#readelf: -s -D --wide 
+#notarget: arc-*-* d30v-*-* dlx-*-* i960-*-* or32-*-* pj*-*-* 
+#notarget: hppa64-*-* i370-*-* i860-*-* ia64-*-* mep-*-* mn10200-*-* 
+# generic linker targets don't support --gc-sections, nor do a bunch of others 
+ 
+#failif 
+#... 
+.*: 0+0 +0 +NOTYPE +GLOBAL +DEFAULT +UND bar 
+#... 
diff --git a/ld/testsuite/ld-elf/pr13177.s b/ld/testsuite/ld-elf/pr13177.s
new file mode 100644
index 0000000..acc36ab
--- /dev/null
+++ b/ld/testsuite/ld-elf/pr13177.s
@@ -0,0 +1,9 @@
+	.section .text.foo,"ax",%progbits 
+	.globl	foo 
+	.type	foo, %function 
+foo: 
+	.byte 0 
+	.section .text.opt_out,"ax",%progbits 
+	.type	opt_out, %function 
+opt_out: 
+	.dc.a bar 
